lib/ui/natives.dart

```diff
@@ -13,10 +13,27 @@ class _Logger {
   static void _printString(String s) native "Logger_PrintString";
 }
 
+// A service protocol extension to schedule a frame to be rendered into the
+// window.
+Future<developer.ServiceExtensionResponse> _scheduleFrame(
+    String method,
+    Map<String, String> parameters) async {
+  // Schedule the frame.
+  window.scheduleFrame();
+  // Always succeed.
+  return new ServiceExtensionResponse.result(JSON.encode({
+    'type': 'Success',
+  }));
+}
+
 void _setupHooks() {
   // Wire up timer implementation that is driven by MojoHandleWatcher.
   VMLibraryHooks.eventHandlerSendData = MojoHandleWatcher.timer;
   VMLibraryHooks.timerMillisecondClock = MojoCoreNatives.timerMillisecondClock;
+  assert(() {
+    // In debug mode, register the schedule frame extension.
+    developer.registerExtension('ext.ui.window.scheduleFrame', _scheduleFrame);
```

Comment:
Went with a separate namespace because the existing 'ext.flutter' extensions are all provided by the framework (package:flutter) and this is provided by 'dart:ui'. I don't have a strong opinion here.